### ISSUE TITLE
library not found on arm64 mac, osx-universal runtime identifier not valid?

### ISSUE THREAD

**Issue by sfuller** [ImGuiNET/ImGui.NET#350]
Hello! Thank you very much for this awesome imgui binding, it was magical seeing imgui up and running in my .net project just by simply installing a .nuget package.

I have recently switch to an arm64 Mac, and have had some issues getting .net to find libcimgui.dylib when using the 1.87.3 nuget package.

It appears that upon restoring the package, `runtimes/osx-universal/native/libcimgui.dylib` is placed in the bin directory, however the runtime doesn't seem to search there.

I have tried adding the `-a universal` arguments to `dotnet run`, but this complains that osx-universal is not a supported rid. `dotnet --version` prints `6.0.400` as the sdk version.

Looking at https://docs.microsoft.com/en-us/dotnet/core/rid-catalog, osx-universal doesn't seem documented at all.

I noticed that in recent commits, the .dylib is now being put in `runtimes/osx-arm64/native/libcimgui.dylib`, however this change has not been released to .nuget yet.

**EDIT:** It is worth noting that my current workaround is to copy the binaries inside of `osx-universal/native` to `osx-arm64/native`. Also worth noting that copying the binaries to `osx/native` does _not_ work. Hope this helps!

In summary, is osx-universal an actual rid, and if not, is this going to be fixed in the next version?

**Comment by Shpendicus** [comment]
I have same issue, just worse, I have an x64 intel and he does not seem to find any imgui lib... I have 3 folders even : "
osx-x64" then "osx-universal" and "OS X-arm64" in all of them I have put the .dylib but when I call it in rider (.net 7 project) it crashes with error code 139?!

**Comment by ashifolfi** [comment]
Linux has this issue as well! I had to place libcimgui.so inside the usr/lib folder to get it to work.
as for Mac I was able to just place it inside the same directory as the executable it compiles and it works fine.


it seems the runtimes folder only works correctly in windows...

**Comment by Shpendicus** [comment]
> Linux has this issue as well it seems! I had to place libcimgui.so inside the usr/lib folder to get it to work. as for Mac I was able to just place it inside the same directory as the executable it compiles and it works fine.
> 
> it seems the runtimes folder only works correctly in windows...

very annoying

**Comment by GaryHuan9** [comment]
The `dotnet` documentation on this topic is horrendous, but I have managed to find some resources related to this issue:

https://github.com/NuGet/Home/issues/6645
https://learn.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks
https://habr.com/en/post/565908/

**Comment by sfuller** [comment]
In ImGui.NET version 1.88, The universal libcimgui.dylib is now placed in the osx/native directory (instead of osx-universal/native).

.NET 6  seems to recognize `osx` as a universal rid, and this is working now out of the box for me without any file copying workarounds.
